SingularityService/src/test/java/com/hubspot/singularity/data/ValidatorTest.java

```diff
@@ -28,4 +48,100 @@ public void testCronExpressionHandlesDayIndexing() {
     Assert.assertEquals("0 0 12 ? * MON-WED", validator.getQuartzScheduleFromCronSchedule("0 12 * * MON-WED"));
   }
 
+  @Test(expected = WebApplicationException.class)
+  public void itForbidsBracketCharactersInDeployIds() throws Exception {
+    final String badDeployId = "deployKey[[";
+
+    SingularityDeploy singularityDeploy = new SingularityDeploy(badDeployId, badDeployId,
```

Comment:
There are `SingularityDeployBuilder` and `SingularityRequestBuilder` methods for this so that you don't need to use the giant constructor. For example you could do:
`SingularityDeploy.newBuilder("requestId", "deployId")`

Fix that up and I think this PR is all set 👍